MIP/mip-draft-FFS/README.md

```diff
@@ -0,0 +1,229 @@
+
+
+**Postconfirmation**. At certain intervals L2-finality certificates will also be published to L1. The L1 contract will verify the L2-finality certificate. This provides an L1-protected _postconfirmation_ that the L2-block (or a batch of L2-blocks) is indeed confirmed. This additional anchoring mechanism increases the security of the L2-confirmation as it locks in the L2-confirmation, reduces the risk of long range attacks and provides a way to slash validators that have attested for invalid blocks.
+
+**Slashing**. The security of the mechanism relies on a PoS protocol. Each validator has to stake some assets, and if they are malicious they _should_ be slashed.
+The condition for slashing may be met by several quiteria, and not all slashing conditions may be used:
+- equivocate (send a different vote to different validators or users)
+- vote :white_check_mark: for an invalid block 
+- vote :x: for a valid block
+
+### Main challenges
+
+To achieve crypto-economically secured Fast-Finality, we need to solve the following problems:
+
+1. design a _staking_ mechanism for the validators to stake assets, distribute rewards and manage slashing
+1. _define and verify_ the threshold (e.g. 2/3 of validators attest :white_check_mark:) for L2-finality
+1. _communicate_ the L2-finality status.
+
+In addition the following may require separate discussion, as it is a different procedure (postconfirmations are handled in smart contracts on L1, whereas L2-finality is handled off-chain)
+
+4. _define and verify_ the threshold (e.g. 2/3 of validators attest :white_check_mark:) for postconfirmation.
+
+
+### Components
+
+#### Staking (addresses 1. and 2.)
+
+The staking mechanism is implemented in a contract on L1.
+This contract provides the following functionalities:
+
+- join: a new validator can join the set of validators by staking some assets
+- exit: a validator exits and  get their stakes back
+- vote: receive a vote or a set of votes, verify the integrity of the votes (signatures) and the minimum threshold (e.g. 2/3)
+
+#### Generate an L2-finality certificate (addreses 3.)
+
+To ensure that the L2-finality status is made available to third-parties, we may publish our _proof_  (2/3 of attestations :white_check_mark:) to a data availability layer and get a _certificate_ that the proof is available.  
+This DA layer should offer a reliable _mempool_ for example as described [in this paper](https://arxiv.org/pdf/2105.11827).
+
+#### Handle postconfirmations (addresses 4.)
+
+The L1 contract will verify the L2-finality certificate. If the certificate is correct the block (or sequence of blocks) are _postconfirmed_. This requires handling who should send the certificate to the L1 contract, and how to verify the certificate.
+
+
+## Reference Implementation
```

Comment:
the details on reference implementation seem to better suitable for a MIP on L2-finality (L2) and an MIP on postconfirmation (L1). but we can commit the proposal in this MIP and then move them to the respective MIP after.

Reply:
We can move them to the list of detailed plan and implementation?

Reply:
its ok here for now :)